src/ray/raylet/scheduling/cluster_task_manager.cc

```diff
@@ -195,58 +231,51 @@ void ClusterTaskManager::Heartbeat(bool light_heartbeat_enabled,
     RAY_CHECK(false) << "TODO";
   } else {
     // TODO (Alex): Implement the 1-CPU task optimization.
-    for (const auto &work : tasks_to_schedule_) {
-      const auto &task = std::get<0>(work);
+    for (const auto &pair : tasks_to_schedule_) {
+      const auto &scheduling_class = pair.first;
       const auto &resources =
-          task.GetTaskSpecification().GetRequiredResources().GetResourceMap();
+          TaskSpecification::GetSchedulingClassDescriptor(scheduling_class)
+              .GetResourceMap();
+      const auto &queue = pair.second;
+      const auto &count = queue.size();
 
       auto by_shape_entry = resource_load_by_shape->Add();
 
       for (const auto &resource : resources) {
         // Add to `resource_loads`.
         const auto &label = resource.first;
         const auto &quantity = resource.second;
-        const auto &entry = resource_loads->find(label);
-        if (entry == resource_loads->end()) {
-          (*resource_loads)[label] = quantity;
-        } else {
-          (*resource_loads)[label] = entry->second + quantity;
-        }
+        (*resource_loads)[label] = quantity * count;
```

Comment:
Isn't this overwriting instead of accumulating into resource_loads?

Reply:
whoops, yeah. modified the test case to catch this.